### PR TITLE
Promote TestProject to lib/

### PR DESCRIPTION
**Pull request by Morriar**
This abstraction has proven very useful and practical to test `spoom` and other sorbet/spoom related tools.

Let's promote it to the lib so it can be reused to test clients.

### CODE REVIEW

File: lib/spoom/test/project.rb [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+# frozen_string_literal: true
+
+require "fileutils"
+require "open3"
+require "pathname"
+
+require_relative "../git"
+
+module Spoom
+  module Test
```
[the remark this answers is not in the document]
Comment (paracycle) [reply lib/spoom/test/project.rb:11]: If changing the name, we need to change the directory name as well.

File: test/test_helper.rb [hunk 2]
```diff
@@ -2,19 +2,18 @@
 # frozen_string_literal: true
 
 $LOAD_PATH.unshift(File.expand_path("../../lib", __FILE__))
-require "spoom"
 
-require "test_project"
+require "spoom"
```
Comment (paracycle) [inline test/test_helper.rb:6]:
```suggestion
require "spoom"
require "spoom/test_helpers/project"
```

File: test/test_helper.rb [hunk 3]
```diff
@@ -2,19 +2,18 @@
...
   module TestHelper
     extend T::Sig
 
     TEST_PROJECTS_PATH = "/tmp/spoom/tests"
 
-    sig { params(name: String).returns(TestProject) }
+    sig { params(name: String).returns(Test::Project) }
     def spoom_project(name)
-      project = TestProject.new("#{TEST_PROJECTS_PATH}/#{name}")
+      project = Test::Project.new("#{TEST_PROJECTS_PATH}/#{name}")
```
Comment (paracycle) [inline test/test_helper.rb:16]:
```suggestion
      project = TestHelpers::Project.new("#{TEST_PROJECTS_PATH}/#{name}")
```

**Comment by Morriar** [comment]
Reworked with TestHelpers module and dir.

**Review by KaanOzkan — APPROVED** [review]

**Review by paracycle — APPROVED** [review]

**Pull request merged**